### PR TITLE
First cut at a python notebook to validate inputs.

### PR DESCRIPTION
**Pull request by gbggrant**
Working for sample_set now.

### CODE REVIEW

File: scripts/variantstore/InputValidation.ipynb [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+    "            if (gvcf_index_field in entity['attributes']):\n",
+    "                reblocked_gvcf_index = entity['attributes'][gvcf_index_field]\n",
+    "\n",
+    "            if (reblocked_gvcf is not None) and (reblocked_gvcf_index is not None):\n",
+    "                reblocked_gvcf_name = reblocked_gvcf.split('/')[-1]\n",
+    "                expected_reblocked_gvcf_index_name = reblocked_gvcf_name + \".tbi\"\n",
+    "                reblocked_gvcf_index_name = reblocked_gvcf_index.split('/')[-1]\n",
+    "                if (reblocked_gvcf_index_name != expected_reblocked_gvcf_index_name):\n",
+    "                    errors_seen = True\n",
+    "                    print(\"ERROR: Did not find expected index file (named: ''\" + expected_reblocked_gvcf_index_name + \n",
```
Comment (mcovarr) [inline scripts/variantstore/InputValidation.ipynb:209]: paren and double single quote issues

File: scripts/variantstore/InputValidation.ipynb [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+    "                         'reblocked_gvcf_index_path',\n",
+    "                         'hg38_reblocked_gvcf_index']\n",
+    "\n",
+    "\n",
+    "samples_in_sample_set = set()\n",
+    "samples_dupes = set()\n",
+    "sample_set = fapi.get_entity(ws_project, ws_name, \"sample_set\", sample_set_id).json()\n",
+    "if (\"attributes\" not in sample_set):\n",
+    "    errors_seen = True\n",
+    "    print(\"ERROR: Looking up \" + sample_set_id + \": ''\" + sample_set[\"message\"] + \"''\")\n",
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/InputValidation.ipynb]: I think I have cleaned those all up.

File: scripts/variantstore/InputValidation.ipynb [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+    "    errors_seen = true\n",
+    "\n",
+    "    \n",
+    "entity_count = entity_types[etype][\"count\"]\n",
+    "\n",
+    "page_size = 1000\n",
+    "num_pages = int(math.ceil(float(entity_count) / page_size))\n",
+    "\n",
+    "# get entities by page where each page has page_size # of rows using API call\n",
+    "#print('Getting all {num_pages} pages of entity data.')\n",
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/InputValidation.ipynb]: Gone

File: scripts/variantstore/InputValidation.ipynb [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+    "for page in tqdm(range(1, num_pages + 1)):\n",
+    "    page_of_entitites = fapi.get_entities_query(ws_project, ws_name, etype, page=page,\n",
+    "                                       page_size=page_size).json()#, sort_direction='asc',\n",
+    "#                                        filter_terms=attribute_names).json()\n",
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/InputValidation.ipynb]: Gone too.

File: scripts/variantstore/InputValidation.ipynb [hunk 5]
```diff
@@ -0,0 +1,275 @@
...
+    "if (len(samples_dupes) > 0):\n",
+    "    errors_seen = True\n",
+    "    print(\"ERROR: Found \" + str(len(samples_dupes)) + \" duplicate sample_ids: \")\n",
+    "    print(list(dict.fromkeys(samples_dupes)))\n",
+    "\n",
+    "print(\"Found \" + str(len(samples_in_sample_set)) + \" samples in sample_set '\" + sample_set_id + \"'\")\n",
+    "\n",
+    "# Inspect samples table - determine possibe names for reblocked_gvcfs.\n",
+    "etype = 'sample'\n",
+    "entity_types = fapi.list_entity_types(ws_project, ws_name).json()\n",
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/InputValidation.ipynb]: yes, good suggestion - incorporated in latest commit.

File: scripts/variantstore/InputValidation.ipynb [hunk 6]
```diff
@@ -0,0 +1,281 @@
...
+    "\n",
+    "    print(f\"Found {str(len(samples_in_sample_set))} samples in sample_set '{sample_set_id}'\")\n",
+    "\n",
+    "if (not errors_seen):\n",
+    "    # Inspect sample table - determine possible names for reblocked_gvcfs.\n",
+    "    gvcf_field, error_seen = get_field_name(reblocked_gvcf_fields, attribute_names, \"reblocked gvcf\")\n",
+    "    if (error_seen):\n",
+    "        errors_seen = True\n",
+    "\n",
+    "    # Inspect sample table - determine possible names for reblocked_gvcf indices.\n",
```
[the remark this answers is not in the document]
Comment (gbggrant) [reply scripts/variantstore/InputValidation.ipynb:182]: I go back and forth on that.

**Review by mcovarr — APPROVED** [review]

File: scripts/variantstore/InputValidation.ipynb [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Input Validation\n",
+    "\n",
+    "This python notebook is intended to allow you to quickly validate the inputs for a Joint Call Set.\n",
+    "To run it:\n",
+    "\n",
+    "Define the variable `sample_set_id` (below) to the name of the sample_set (in the current workspace) containing the list of samples to process\n",
```
Comment (rsasch) [inline scripts/variantstore/InputValidation.ipynb:12]: nit
```suggestion
    "Define the variable `sample_set_id` (below) to the name of the sample_set (in the current workspace) containing the list of samples to process.\n",
```

File: scripts/variantstore/InputValidation.ipynb [hunk 8]
```diff
@@ -0,0 +1,281 @@
...
+    "\n",
+    "The notebook will validate that:\n",
+    "- the sample set that you have listed is found\n",
+    "- there are no duplicate samples in the sample set\n",
+    "- there are no empty sample names in the sample set\n",
+    "- each sample has a corresponding reblocked_gvcf index\n",
```
Comment (rsasch) [inline scripts/variantstore/InputValidation.ipynb:18]: maybe
```suggestion
    "- each sample has a reblocked gVCF and a corresponding reblocked gVCF index\n",
```

**Review by rsasch — APPROVED** [review]
Looks good except for some small nits (and one more for good measure, "sample set" or "sample_set" not both).

**Pull request merged**